component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -178,6 +178,12 @@ public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bod
 
         String headerName = IdentityUtil.getProperty(CommonConstants.MTLS_AUTH_HEADER);
         if (clientIdExistsAsParam(bodyParams)) {
+            // If the Private key JWT authenticator was hit previously, then the MTLS authenticator should
+            // not authenticate the client.
+            if (CommonConstants.AUTHENTICATOR_TYPE_PK_JWT.equals((String)
+                    context.getParameter(CommonConstants.AUTHENTICATOR_TYPE_PARAM))) {
```

Comment:
lets add a debug log here.

Reply:
Fixed in 159511f58156995bb4ad0e05d019c4349ee04bcd